node/bft/src/gateway.rs

```diff
@@ -117,6 +117,8 @@ pub struct Gateway<N: Network> {
     /// prevent simultaneous "two-way" connections between two peers (i.e. both nodes simultaneously
     /// attempt to connect to each other). This set is used to prevent this from happening.
     connecting_peers: Arc<Mutex<IndexSet<SocketAddr>>>,
+    /// The cached view of connected validators validators.
+    cached_validator_view: Arc<RwLock<IndexMap<SocketAddr, Address<N>>>>,
```

Comment:
```suggestion
    cached_validator_view: Arc<Mutex<HashMap<SocketAddr, Address<N>>>>,
```
since we only use it for writing, a `Mutex` is faster, and if we don't care about insertion order, a `HashMap` is leaner